public/components/JoinDatasetsForm.vue

```diff
@@ -198,7 +198,7 @@ export default Vue.extend({
         .then((tableData) => {
           this.pending = false;
           this.showJoinSuccess = true;
-          this.previewTableData = tableData;
+          this.previewTableData = Object.seal(tableData);
```

Comment:
Sorry to be pedantic about this, but can you add a small comment on why you use `Object.seal()`. Usually, people use this method to make the object non-mutable, but here it's to circumvent the bootstrap-component.

Reply:
I can add that really quick. In a sense here, it's also an issue of immutability, just for different reasons, but yes, I need to make it clear so it never is callously reverted.

Reply:
Comments added 👍